token/program-2022-test/tests/confidential_transfer.rs

```diff
@@ -2106,3 +2139,688 @@ async fn ct_transfer_with_fee_memo() {
+        ];
+
+        let tx = Transaction::new_signed_with_payer(
+            &instructions,
+            Some(&ctx.payer.pubkey()),
+            &[&ctx.payer, &context_state_account],
+            ctx.last_blockhash,
+        );
+        ctx.banks_client.process_transaction(tx).await.unwrap();
+    }
+
+    token
+        .confidential_transfer_withdraw(
+            &alice_meta.token_account,
+            &alice.pubkey(),
+            Some(&context_state_account.pubkey()),
+            0,
+            decimals,
+            None,
+            &alice_meta.elgamal_keypair,
+            &alice_meta.aes_key,
+            &[&alice],
+        )
+        .await
+        .unwrap();
+
+    // attempt to create an account with a wrong proof type context state
+    let bob_meta = ConfidentialTokenAccountMeta::new(&token, &bob).await;
+    let context_state_account = Keypair::new();
+
+    {
+        let context_state_authority = Keypair::new();
+        let space = size_of::<ProofContextState<PubkeyValidityProofContext>>();
+
+        let instruction_type = ProofInstruction::VerifyPubkeyValidity;
+
+        let context_state_info = ContextStateInfo {
+            context_state_account: &context_state_account.pubkey(),
+            context_state_authority: &context_state_authority.pubkey(),
+        };
+
+        let proof_data =
+            confidential_transfer::instruction::PubkeyValidityData::new(&bob_meta.elgamal_keypair)
+                .unwrap();
+
+        let mut ctx = context.context.lock().await;
+        let rent = ctx.banks_client.get_rent().await.unwrap();
+
+        let instructions = vec![
+            system_instruction::create_account(
+                &ctx.payer.pubkey(),
+                &context_state_account.pubkey(),
+                rent.minimum_balance(space),
+                space as u64,
+                &zk_token_proof_program::id(),
+            ),
+            instruction_type.encode_verify_proof(Some(context_state_info), &proof_data),
+        ];
+
+        let tx = Transaction::new_signed_with_payer(
+            &instructions,
+            Some(&ctx.payer.pubkey()),
+            &[&ctx.payer, &context_state_account],
+            ctx.last_blockhash,
+        );
+        ctx.banks_client.process_transaction(tx).await.unwrap();
+    }
+
+    let err = token
+        .confidential_transfer_withdraw(
+            &bob_meta.token_account,
+            &bob.pubkey(),
+            Some(&context_state_account.pubkey()),
+            0,
+            decimals,
+            None,
+            &bob_meta.elgamal_keypair,
+            &bob_meta.aes_key,
+            &[&bob],
+        )
+        .await
+        .unwrap_err();
+
+    assert_eq!(
+        err,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InstructionError(0, InstructionError::InvalidArgument,)
+        )))
+    );
+}
+
+#[tokio::test]
+async fn confidential_transfer_transfer_with_proof_context() {
```

Comment:
Can you add a test case for transfer with fee?

Reply:
Since this involves the confidential transfer fee extension, I will add the tests on a subsequent PR when I update all the fee component to Solana 1.16.